client/my-sites/checkout/composite-checkout/lib/filter-appropriate-payment-methods.ts

```diff
@@ -23,16 +23,9 @@ export default function filterAppropriatePaymentMethods( {
 		.filter( ( methodObject ) => {
```

Comment:
nit: I'd prefer if the `filterAppropriatePaymentMethods` function was implemented like this:

```javascript
const isPurchaseFree = responseCart.total_cost_integer === 0 || doesPurchaseHaveFullCredits( responseCart );

if ( isPurchaseFree ) {
	return paymentMethodObjects.filter( ( methodObject ) => methodObject.id === 'free-purchase' );
}

return paymentMethodObjects
	.filter( ( methodObject ) => {
		const slug = readCheckoutPaymentMethodSlug( methodObject.id );
		if ( ! slug || slug === 'free-purchase' ) {
			return false;
		}
		return isPaymentMethodEnabled( slug, allowedPaymentMethods, countryCode );
	} );
```
do you see any issues? I feel that the intention of the code is clearer this way.

Reply:
Looks good! Yes, I agree that it will be more clear this way, although the next thing I'm going to do is put this logic on the backend, so we should soon be able to remove the `isPurchaseFree` logic here entirely.

Reply:
Sounds good! 😄